typings/index.d.ts

```diff
@@ -378,6 +379,239 @@ declare module 'discord.js' {
+		};
+		browser: boolean;
+		DefaultOptions: ClientOptions;
+		UserAgent: string | null;
+		Endpoints: {
+			botGateway: string;
+			invite: (root: string, code: string) => string;
+			CDN: (root: string) => {
+				Asset: (name: string) => string;
+				DefaultAvatar: (id: string | number) => string;
+				Emoji: (emojiID: string, format: 'png' | 'gif') => string;
+				Avatar: (userID: string | number, hash: string, format: AllowedImageFormat, size: number) => string;
+				Icon: (userID: string | number, hash: string, format: AllowedImageFormat, size: number) => string;
+				AppIcon: (userID: string | number, hash: string, format: AllowedImageFormat, size: number) => string;
+				AppAsset: (userID: string | number, hash: string, format: AllowedImageFormat, size: number) => string;
+				GDMIcon: (userID: string | number, hash: string, format: AllowedImageFormat, size: number) => string;
+				Splash: (userID: string | number, hash: string, format: AllowedImageFormat, size: number) => string;
+			};
+		};
+		WSCodes: {
+			1000: 'Connection gracefully closed';
+			4004: 'Tried to identify with an invalid token';
+			4010: 'Sharding data provided was invalid';
+			4011: 'Shard would be on too many guilds if connected';
+		};
+		Events: {
+			RATE_LIMIT: 'rateLimit';
+			READY: 'ready';
+			RESUMED: 'resumed';
+			GUILD_CREATE: 'guildCreate';
+			GUILD_DELETE: 'guildDelete';
+			GUILD_UPDATE: 'guildUpdate';
+			GUILD_UNAVAILABLE: 'guildUnavailable';
+			GUILD_AVAILABLE: 'guildAvailable';
+			GUILD_MEMBER_ADD: 'guildMemberAdd';
+			GUILD_MEMBER_REMOVE: 'guildMemberRemove';
+			GUILD_MEMBER_UPDATE: 'guildMemberUpdate';
+			GUILD_MEMBER_AVAILABLE: 'guildMemberAvailable';
+			GUILD_MEMBER_SPEAKING: 'guildMemberSpeaking';
+			GUILD_MEMBERS_CHUNK: 'guildMembersChunk';
+			GUILD_INTEGRATIONS_UPDATE: 'guildIntegrationsUpdate';
+			GUILD_ROLE_CREATE: 'roleCreate';
+			GUILD_ROLE_DELETE: 'roleDelete';
+			GUILD_ROLE_UPDATE: 'roleUpdate';
+			GUILD_EMOJI_CREATE: 'emojiCreate';
+			GUILD_EMOJI_DELETE: 'emojiDelete';
+			GUILD_EMOJI_UPDATE: 'emojiUpdate';
+			GUILD_BAN_ADD: 'guildBanAdd';
+			GUILD_BAN_REMOVE: 'guildBanRemove';
+			CHANNEL_CREATE: 'channelCreate';
+			CHANNEL_DELETE: 'channelDelete';
+			CHANNEL_UPDATE: 'channelUpdate';
+			CHANNEL_PINS_UPDATE: 'channelPinsUpdate';
+			MESSAGE_CREATE: 'message';
+			MESSAGE_DELETE: 'messageDelete';
+			MESSAGE_UPDATE: 'messageUpdate';
+			MESSAGE_BULK_DELETE: 'messageDeleteBulk';
+			MESSAGE_REACTION_ADD: 'messageReactionAdd';
+			MESSAGE_REACTION_REMOVE: 'messageReactionRemove';
+			MESSAGE_REACTION_REMOVE_ALL: 'messageReactionRemoveAll';
+			USER_UPDATE: 'userUpdate';
+			USER_NOTE_UPDATE: 'userNoteUpdate';
+			USER_SETTINGS_UPDATE: 'clientUserSettingsUpdate';
+			PRESENCE_UPDATE: 'presenceUpdate';
+			VOICE_STATE_UPDATE: 'voiceStateUpdate';
+			VOICE_BROADCAST_SUBSCRIBE: 'subscribe';
+			VOICE_BROADCAST_UNSUBSCRIBE: 'unsubscribe';
+			TYPING_START: 'typingStart';
+			TYPING_STOP: 'typingStop';
```

Comment:
While being in the constants this event is not actually emitted from within d.js.